### PR TITLE
Add new test routines and update to calc_K()

### PR DESCRIPTION
**Pull request by dm807cam**
This PR adds:
- An updated testthat routine
- New Github Action to compare calculated Ks against check values
- An update to calc_K() to ensure that pyMyAMI is not called repeatedly in calc_Ks()

### CODE REVIEW

File: r/R/main.R [hunk 1]
```diff
@@ -104,11 +103,27 @@ calc_Ks <- function(k_list, TC=25, S=35, Mg=0.0528171, Ca=0.0102821, P=NULL, MyA
...
+  
+  # Calculate correction factor with MyAMI
+  if(MyAMI_calc == TRUE){ 
+    Fcorr = pymyami$calc_Fcorr(Sal=S, TempC=TC, Mg=Mg, Ca=Ca)
+  } else {
+    Fcorr = pymyami$approximate_Fcorr(Sal=S, TempC=TC, Mg=Mg, Ca=Ca)
+  }
+    
+  # Apply correction
+  KF = Fcorr[[k]]
```
Comment (oscarbranson) [inline r/R/main.R:121]: At written, this block applies a single Fcorr to all the Ks, whereas the Fcorr for each K is different. This needs to loop through all the Ks and apply each Fcorr to each K separately.

This is causing crosscheck to fail.

File: r/R/main.R [hunk 2]
```diff
@@ -60,19 +59,21 @@ calc_K <- function(k, TC=25, S=35, Mg=0.0528171, Ca=0.0102821, P=NULL, MyAMI_cal
...
   
-  # Apply correction
-  KF = Fcorr[[k]]
-  if(!is.null(KF)){
-    check_KF = ifelse(KF != 0, KF, 1)
-    K = K * check_KF
+    # Apply correction
+    KF = Fcorr[[k]]
+    if(!is.null(KF)){
+      check_KF = ifelse(KF != 0, KF, 1)
```
Comment (oscarbranson) [inline r/R/main.R:72]: I'm also not sure this line is necessary... pymyami should never produce a case where KF == 0... I think?

File: r/R/main.R (hunk 1 above)
Comment (dm807cam) [reply r/R/main.R]: Yes, must have missed that. Corrected in new PR.

File: r/R/main.R (hunk 2 above)
Comment (dm807cam) [reply r/R/main.R:72]: True, that is a remain from an earlier error tracing. Removed in new PR.

**Pull request merged**